sdk/keyvault/Azure.Security.KeyVault.Shared/src/ChallengeBasedAuthenticationPolicy.cs

```diff
@@ -93,6 +93,21 @@ private async ValueTask<bool> AuthorizeRequestOnChallengeAsyncInternal(HttpMessa
 
             string authority = GetRequestAuthority(message.Request);
             string scope = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "resource");
+
+            string error = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "error");
+            string claims = null;
+
+            if (error != null)
+            {
+                // The challenge response contained an error.
+                string base64Claims = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "claims");
+
+                if (error == "insufficient_claims" && base64Claims != null)
+                {
+                    claims = System.Text.Encoding.UTF8.GetString(Convert.FromBase64String(base64Claims));
+                }
+            }
```

Comment:
```suggestion
            if (error == "insufficient_claims")
            {
                 claims = AuthorizationChallengeParser.GetChallengeParameterFromResponse(message.Response, "Bearer", "claims") switch
                {
                    { Length: 0 } => null,
	        	    string enc => System.Text.Encoding.UTF8.GetString(Convert.FromBase64String(enc))
	        	};
            }
```

I think you can simplify a bit by avoiding the double check on `error` and conditionalizing the `claims` assignment.